tests/driver/runner_test.py

```diff
@@ -15,24 +16,36 @@ def runner_function(es, params):
 
         runner.register_runner(operation_type="unit_test", runner=runner_function)
         returned_runner = runner.runner_for("unit_test")
-        self.assertIsInstance(returned_runner, runner.DelegatingRunner)
+        self.assertIsInstance(returned_runner, runner.SingleClusterDelegatingRunner)
         self.assertEqual("user-defined runner for [runner_function]", repr(returned_runner))
 
-    def test_runner_class_with_context_manager_should_be_registered_as_is(self):
-        class UnitTestRunner:
-            def __enter__(self):
+    def test_runner_class_with_context_manager_should_be_wrapped_with_context_manager_enabled(self):
+        class UnitTestContextManagerRunner:
+            def __call__(self, *args):
                 return self
 
-            def __call__(self, *args):
-                pass
+            def __enter__(self):
+                self.fp = io.StringIO("many\nlines\nin\na\nfile")
+                return self.fp
 
             def __exit__(self, exc_type, exc_val, exc_tb):
-                return False
+                return self.fp.close()
```

Comment:
I think the implementation should be:

```python
self.fp.close()
return False
```

The return value indicates whether a potential exception has been handled (indicated by `return True`). As we don't  do anything here, `return False` is warranted IMHO.

Reply:
Agreed. Will address.